addons/l10n_ke_edi_tremol/models/account_move.py

```diff
@@ -0,0 +1,227 @@
+                    errors.append(_("On line %s, a product with a HS Code and HS Name must be selected, since the tax is 8%%, 0%% or exempt.", line.name))
+        return errors
+
+    def _l10n_ke_validate_taxes(self):
+        """ Returns list of misconfigurations on the taxes of the move that would cause rejection by the KRA """
+        self.ensure_one()
+        errors = []
+        for tax in self.invoice_line_ids.tax_ids:
+            if tax.amount not in (16, 8, 0):
+                errors.append(_("Tax '%s' is used, but only taxes of 16%%, 8%%, 0%% or Exempt can be sent. Please reconfigure or change the tax.", tax.name))
+        return errors
+
+
+    # -------------------------------------------------------------------------
+    # SERIALISERS
+    # -------------------------------------------------------------------------
+
+    def _l10n_ke_cu_open_invoice_message(self):
+        """ Serialise the required fields for opening an invoice
+
+        :returns: a list containing one byte-string representing the <CMD> and
+                  <DATA> of the message sent to the fiscal device.
+        """
+        headquarter_address = (self.commercial_partner_id.street or '') + (self.commercial_partner_id.street2 or '')
+        customer_address = (self.partner_id.street or '') + (self.partner_id.street2 or '')
+        postcode_and_city = self.partner_id.zip or '' + ' ' +  self.partner_id.city or ''
+        invoice_elements = [
+            b'1',                                                   # Reserved - 1 symbol with value '1'
+            b'     0',                                              # Reserved - 1 symbol with value ‘     0’
+            b'0',                                                   # Reserved - 1 symbol with value '0'
+            b'1' if self.move_type == 'out_invoice' else b'A',      # 1 symbol with value '1' (new invoice), 'A' (credit note), or '@' (debit note)
+            self._l10n_ke_fmt(self.commercial_partner_id.name, 30), # 30 symbols for Company name
+            self._l10n_ke_fmt(self.commercial_partner_id.vat, 14),  # 14 Symbols for the client PIN number
+            self._l10n_ke_fmt(headquarter_address, 30),             # 30 Symbols for customer headquarters
+            self._l10n_ke_fmt(customer_address, 30),                # 30 Symbols for the address
+            self._l10n_ke_fmt(postcode_and_city, 30),               # 30 symbols for the customer post code and city
+            self._l10n_ke_fmt('', 30),                              # 30 symbols for the exemption number
+        ]
+        if self.move_type in ('out_refund', 'debit_note'):
+            invoice_elements.append(self._l10n_ke_fmt(self.reversed_entry_id.l10n_ke_cu_invoice_number, 19)), # 19 symbols for related invoice number
+        invoice_elements.append(self._l10n_ke_fmt(self.name, 15))   # 15 symbols for trader system invoice number
```

Comment:
Is there a way to have [15:] instead of [:15] for this?  (the number info is mostly in the end...)